src/v/archival/tests/service_test.cc

```diff
@@ -139,15 +139,18 @@ FIXTURE_TEST(test_segment_upload, archiver_fixture) {
     auto topic = model::topic("topic_3");
     auto ntp = model::ntp(test_ns, topic, model::partition_id(0));
 
-    const char* manifest_url
-      = "/c0000000/meta/test-namespace/topic_3/0_2/manifest.json";
+    model::revision_id partition_rev{get_next_partition_revision_id().get()};
+    std::string manifest_path = fmt::format(
+      "/d0000000/meta/test-namespace/topic_3/0_{}/manifest.json",
```

Comment:
Good point about the hash -- I've updated this to calculate that dynamically as well.

Using a static revision ID doesn't work, because later when doing `add_topic_with_archival_enabled`, the normal topic creation code runs, which uses the controller log offset as revision ID, so will disagree with any hardcoded revision IDs in the test when there are additional messages in the controller log.